src/components/application_manager/src/resumption/resume_ctrl_impl.cc

```diff
@@ -214,12 +214,20 @@ bool ResumeCtrlImpl::RestoreAppHMIState(ApplicationSharedPtr application) {
             "High-bandwidth transport not available, app will resume into : "
                 << saved_hmi_level);
       }
-
       const bool app_hmi_state_is_set =
           SetAppHMIState(application, saved_hmi_level, true);
+      size_t restored_widgets = 0;
       if (app_hmi_state_is_set &&
           application->is_app_data_resumption_allowed()) {
-        RestoreAppWidgets(application, saved_app);
+        restored_widgets = RestoreAppWidgets(application, saved_app);
+      }
+      const bool does_app_with_same_level_exist =
+          application_manager_.IsAppTypeExistsInFullOrLimited(application);
+      if (does_app_with_same_level_exist && restored_widgets == 0) {
```

Comment:
@JackLivio I would recommend to swap comparables : `restored_widgets == 0` to `0 == restored_widgets`